radix-engine/src/kernel/kernel.rs

```diff
@@ -495,8 +492,24 @@ where
         self.current_frame.depth
     }
 
-    fn kernel_get_current_actor(&self) -> Option<Actor> {
-        self.current_frame.actor.clone()
+    fn kernel_get_current_actor(&mut self) -> Option<Actor> {
+        let actor = self.current_frame.actor.clone();
+        if let Some(actor) = &actor {
+            match actor {
+                Actor::Method {
+                    global_address: Some(address),
+                    ..
+                } => {
+                    self.current_frame.add_ref(
+                        RENodeId::GlobalObject(*address),
+                        RENodeVisibilityOrigin::Normal,
+                    );
+                }
```

Comment:
Should we take the approach regarding package address?

Reply:
good point

Reply:
We can then remove some of the hacks in `scrypto_interpreter`.